tools/DeleteModules.ps1

```diff
@@ -0,0 +1,193 @@
+        $previousDependencies = Find-Module -Name $query -Repository $repoName -RequiredVersion $azureRmPrevious.Version -IncludeDependencies
+        $currentDependencies | ForEach-Object {
+            $CDModule = $_
+            $versionChanged = $true
+            $previousDependencies | ForEach-Object {
+                if (($_.Name -eq $CDModule.Name) -and ($_.Version -eq $CDModule.Version))
+                {
+                    $versionChanged = $false
+                }
+            }
+
+            if ($versionChanged)
+            {
+                $targets += $CDModule
+            }
+        }
+
+        return $targets
+    }
+}
+
+function Get-DependentModules
+{
+    param(
+        [string]$repoName,
+        [string]$moduleName,
+        [string]$moduleVersion,
+        $allModules,
+        $azureModules
+    )
+
+    $dependencies = @()
+    $azureModules | ForEach-Object {
+        $targetName = $_.Name
+        $moduleNotBeingDeleted = $true
+        $allModules | ForEach-Object {
+            if ($_.Name -eq $targetName)
+            {
+                $moduleNotBeingDeleted = $false
+            }
+        }
+
+        $_.Dependencies | ForEach-Object {
+            $dependencyName = $_.Name
+
+            if ($moduleNotBeingDeleted -and ($moduleName -eq $dependencyName))
+            {
+                $allTargetVersions = Find-Module -Name $targetName -Repository $repoName -AllVersions
+                $allTargetVersions | ForEach-Object {
+                    $targetVersion = $_.Version
+                    $_.Dependencies | ForEach-Object {
+                        if (($moduleName -eq $_.Name) -and (($_.MinimumVersion -eq $moduleVersion) -or ($_.RequiredVersion -eq $moduleVersion)))
+                        {
+                            $dependencies += "`"$targetName $targetVersion`","
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if ($dependencies.Count -ne 0)
+    {
+        Write-Warning "$moduleName $moduleVersion is a dependency for $dependencies the module(s) will have an orphaned dependency."
+    }
+}
+
+function Get-ApiKey
+{
+	param([string]$repoName)
+    
+    $vaultKey="PSTestGalleryApiKey"
+    if ($repoName -eq "PSGallery")
+    {
+        $vaultKey = "PowerShellGalleryApiKey"
+    }
+    
+    $context = $null
+    try {
+        $context = Get-AzureRMContext
+    } catch {}
+
+    if ($context -eq $null)
+    {
+        Add-AzureRMAccount
+    }
+
+    $secret = Get-AzureKeyVaultSecret -VaultName kv-azuresdk -Name $vaultKey
+
+    $secret.SecretValueText
+}
+
+if ([string]::IsNullOrEmpty($nugetExe))
+{
+    Write-Verbose "Use default nuget path"
+    $nugetExe =  "$PSScriptRoot\nuget.exe"
+}
+
+if ([string]::IsNullOrEmpty($repoName))
+{
+    Write-Verbose "Deleting from PSGallery"
+    $repoName = "PSGallery"
+}
+
+if (($scope -eq "ModuleList") -and ($listOfModules -eq $null))
```

Comment:
@maddieclayton can we create two separate parameter sets, one that takes `Scope` and another that takes `ListOfModules`?